content/blogposts/2024/codyception-coding-cody-tools-with-cody.md

```diff
@@ -0,0 +1,159 @@
+
+OK, I'll admit it took me longer than it should to write that title without messing up, but here we are, there that is, and here we go!
+
+So, now that I've successfully confused half of you before I've even started, let's talk about Custom Commands with Cody.
+
+---
+
+### Friends, Coders, Netizens, lend me your snippets!
+
+
+To start with, let's delve into the significance of Custom Commands within the realm of programming and automation. These commands play a pivotal role by affording you the ability to define specific actions tailored to your precise needs. In more elementary terms, they act as a time-saving mechanism, sparing you considerable effort, especially in tasks characterized by repetition. Consider scenarios like utilizing Cody to insert specific types of comments, code editing, feature incorporation, or even language translation for entire files.  In even more simplistic terms, they let you make a button that does a thing, then you can click that button to do that thing an unlimited amount of times without having to type anything again!  Sound good? I bet it does...so let's talk about it in the context of our main man Cody.
+
+The elegance of Cody lies in its seamless facilitation of the creation and integration of these commands. This capability transforms the coding landscape, rendering your journey notably more streamlined and efficient. Cody becomes a reliable companion, adept at executing nuanced coding tasks with dexterity.
+
+In essence, Custom Commands serve as the virtuosos of your coding orchestra, orchestrating intricate symphonies with finesse. As you navigate the coding landscape, allow these commands to be your skilful conductors, ensuring a harmonious and efficient coding experience. 
+
+Are you hungry for more details? Let's put away those cookbooks because we don't need recipes any more!
+
+### Evolving from Recipes to Commands
+
+Gone are the days of being limited to the Recipes tab and submitting pull requests. With Commands, you can effortlessly create and execute reusable prompts without opening the Chat sidebar. This improvement streamlines your workflow and eliminates the need for pull requests, making the process much smoother.
+
+**An example command JSON that would sit comfortably in your cody.json file:**
+
+```json
+    "algorithm-analysis": {
+      "description": "algorithm-analysis",
+      "prompt": "Analyze and document the algorithms used in the code, detailing their efficiency and how they achieve their intended purpose.",
+      "context": {
+        "codebase": true,
+        "selection": true,
+        "currentFile": true
+      }
+```
+
+This would then be called in your editor with a slash command: `/implement-code-analytics-tracking`
```

Comment:
Hmm the slash command is different from the above JSON?

Reply:
Good catch